### DISCUSSION TITLE
使用文档类`sjtuarticle`报错+PDF显示不正确

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [sjtug/SJTUThesis#906] [提问] [answered]
目前编译模板里面的`sjtuthesis`是没有问题的。删除`main.tex`和`setup.tex`后，试着使用文档类`sjtuarticle`。

- 将[sample-article-zh.tex](https://github.com/sjtug/SJTUTeX/blob/v2/sjtutex/doc/sample-article-zh.tex)复制过来，改成`main.tex`；并直接将\input{common-package}中的内容直接复制到`main.tex`里面
- 将[ref.bib](https://github.com/sjtug/SJTUTeX/blob/v2/sjtutex/support/ref.bib)复制过来

直接运行`make all`编译时，会报错（运行多次也是一样）：

> ./main.tex:23: Undefined control sequence.
l.23 \subject
             {期末课程论文} 
   ...
   make: *** [Makefile:24: main.pdf] Error 12

目前可行的方案是：手动运行4次`latexmk`生成PDF（应该是交叉编译的问题），并且奇怪的是，**生成的PDF的关键词位置不正确：出现在首页，而不是在摘要的后面**。

![image](https://user-images.githubusercontent.com/2310958/227875844-b3305dbf-ea6a-4005-87bd-32fb7963b7af.png)

这应该如何解决？

---
OS: Manjaro; Tex: Texlive 2022.62885-3

**Comment by AlexaraWu** [comment, marked as answer]
SJTUTeX 那边 `sjtuarticle` 改版过了，这个仓库里的 `sjtuarticle` 还没有同步更新。
